### DISCUSSION TITLE
Pipeline/Cloud Formation Error

### DISCUSSION THREAD

**Discussion by anne-montoya** [aws/copilot-cli#3922] [Q&A] [answered]
I successfully updated my env (`copilot env deploy`) and our services (`copilot svc deploy`) to get the changes from v1.21
But our pipelines encountered error on deployment stage.

So I also updated our pipelines with `copilot pipeline update` but deployment is still failing on _CloudFormation_, specifically on these 2 resources: **HTTPListenerRuleWithDomain** and **HTTPSListenerRule**

The error for both resources are`CustomResource attribute error: Vendor response doesn't contain Priority key in object`. When I trace the template it got something to do with `HTTPRuleWithDomainPriorityAction` and `HTTPSRulePriorityAction` which uses this `RulePriorityFunction`. Am I missing some permissions for codepipeline?

**Comment by dannyrandall** [comment, marked as answer]
Hey @anne-montoya! Are you using a pipeline created by Copilot? If so, have you updated your `buildspec.yml` to use the latest version of Copilot as well? You'll need to update these lines, changing the `copilot-linux-v1.xx.x` to `copilot-linux-v1.21.0`:
```
      # Download the copilot linux binary.
      - wget -q https://ecs-cli-v2-release.s3.amazonaws.com/copilot-linux-v1.21.0
      - mv ./copilot-linux-v1.21.0 ./copilot-linux
      - chmod +x ./copilot-linux
```

**Reply by anne-montoya** [reply]
Hello @dannyrandall , im getting this error now

`[/Resources/EnvControllerFunction/Type/Code/S3Bucket] 'null' values are not allowed in templates (Service: AmazonCloudFormation; Status Code: 400; Error Code: ValidationError; Request ID: 1e378316-6e8a-4b50-84fa-4573181e21d5; Proxy: null)`

**Reply by dannyrandall** [reply]
Maybe that's similar to https://github.com/aws/copilot-cli/issues/3825? Could you try adding the `--upload-assets` flag to the `copilot svc package` line and `copilot job package` lines?

**Reply by anne-montoya** [reply]
I updated my buildspec.yml, which uses v1.10, to get the changes from v1.21 via `copilot pipeline init`
Thanks @dannyrandall 😄
